File: src/2025-02-27-deepclaude37-rust/javascript/exercises/practice/phone-number/phone-number.js
//
// This is only a SKELETON file for the 'Phone Number' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const clean = (number) => {
  // Check for letters in the input
  if (/[a-zA-Z]/.test(number)) {
    throw new Error('Letters not permitted');
  }

  // Check for punctuation that's not allowed
  if (/[^\d\s\(\)\-\.\+]/.test(number)) {
    throw new Error('Punctuations not permitted');
  }

  // Remove all non-digit characters
  let cleaned = number.replace(/\D/g, '');

  // Validate length and country code
  const length = cleaned.length;
  if (length < 10) {
    throw new Error('Incorrect number of digits');
  }
  if (length > 11) {
    throw new Error('More than 11 digits');
  }
  if (length === 11) {
    if (cleaned[0] !== '1') {
      throw new Error('11 digits must start with 1');
    }
    cleaned = cleaned.substring(1);
  }

  // Validate area code
  if (cleaned[0] === '0') {
    throw new Error('Area code cannot start with zero');
  }
  if (cleaned[0] === '1') {
    throw new Error('Area code cannot start with one');
  }

  // Validate exchange code
  if (cleaned[3] === '0') {
    throw new Error('Exchange code cannot start with zero');
  }
  if (cleaned[3] === '1') {
    throw new Error('Exchange code cannot start with one');
  }

  return cleaned;
};

File: src/2025-02-27-deepclaude37-rust/java/exercises/practice/phone-number/src/main/java/PhoneNumber.java
class PhoneNumber {
    private final String number;

    PhoneNumber(String numberString) {
        // Remove all non-digit characters
        String cleaned = numberString.replaceAll("[^0-9]", "");
        
        // Handle country code if present
        if (cleaned.length() == 11) {
            if (cleaned.startsWith("1")) {
                cleaned = cleaned.substring(1);
            } else {
                throw new IllegalArgumentException("11 digits must start with 1");
            }
        }
        
        // Validate the length
        if (cleaned.length() != 10) {
            throw new IllegalArgumentException("Number must be 10 digits");
        }
        
        // Validate area code (first digit)
        if (cleaned.charAt(0) < '2' || cleaned.charAt(0) > '9') {
            throw new IllegalArgumentException("Area code must start with a digit from 2-9");
        }
        
        // Validate exchange code (fourth digit)
        if (cleaned.charAt(3) < '2' || cleaned.charAt(3) > '9') {
            throw new IllegalArgumentException("Exchange code must start with a digit from 2-9");
        }
        
        this.number = cleaned;
    }

    String getNumber() {
        return number;
    }
}
